.github/workflows/deploy.yml

```diff
@@ -105,6 +105,100 @@ jobs:
           name: ${{ matrix.name }}
           path: ${{ matrix.name }}
 
+  # Notarize starship binaries for MacOS and build notarized pkg installers
+  notarize_and_pkgbuild:
+    runs-on: macos-latest
+    needs: github_build
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target: x86_64-apple-darwin
+            arch: x86_64
+            name: starship-x86_64-apple-darwin.tar.gz
+            pkgname: starship-x86_64-apple-darwin.pkg
+
+          - target: aarch64-apple-darwin
+            arch: aarch64
+            name: starship-aarch64-apple-darwin.tar.gz
+            pkgname: starship-aarch64-apple-darwin.pkg
+
+    env:
+      KEYCHAIN_FILENAME: app-signing.keychain-db
+      KEYCHAIN_ENTRY: AC_PASSWORD
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Notarize | Set up secrets
+      env:
+        APP_CERTIFICATE_BASE64: ${{ secrets.APPLEDEV_APPSIGNKEY_BASE64 }}
+        INSTALL_CERTIFICATE_BASE64: ${{ secrets.APPLEDEV_INSTALLERSIGNKEY_BASE64 }}
+        P12_PASSWORD: ${{ secrets.APPLEDEV_SIGNKEY_PASS }}
+        KEYCHAIN_PASSWORD: ${{ secrets.APPLEDEV_SIGNKEY_PASS }}
+        APPLEID_USERNAME: ${{ secrets.APPLEDEV_ID_NAME }}
+        APPLEID_TEAMID: ${{ secrets.APPLEDEV_TEAM_ID }}
```

Comment:
Out of curiosity, will this be signed as Starship or as you?

Reply:
This is going to be signed by me. We can get signing certificates for at least one other mission control member as well, in case I get kidnapped by cephalopods or hit by a bus, but Apple has some [rather high requirements](https://developer.apple.com/programs/enroll/) for enrolling an organization: in particular, the organization needs to have legal entity status (be capable of sued) and a [DUNS number](https://www.dnb.com/duns-number.html).

Also, I believe the signing certificate I exported was due to expire in 2027, so if it happens in the next couple years, you should have quite a bit of time to work out a replacement :)

Reply:
I'm not at my computer right now, but we may qualify as part of our Open Collective fiscal sponsorship, which qualifies us as a non-profit in the US, which may give us that legal status.

Reply:
Oh interesting. If that's the case, I believe you would have to be the one to do the registration, since another requirement is

> As the person enrolling your organization in the Apple Developer Program, you must have the legal authority to bind your organization to legal agreements. You must be the organization’s owner/founder, executive team member, senior project lead, or an employee with legal authority granted to you by a senior employee.

Once that's done though, I believe individuals should be able to acquire signing keys that work on behalf of the organization.

I don't believe that signing keys from an organization would fundamentally be any different than those from an individual (the organization ID + signing key IDs would have to change in the GH secrets, but that's about it) so switching over if we decide to go for an Apple Dev organization shouldn't be too much of an issue.